src/bidiMapper/domains/context/browsingContextImpl.ts

```diff
@@ -320,12 +301,26 @@ export class BrowsingContextImpl {
         if (this.id !== params.frame.id) {
           return;
         }
+        const timestamp = BrowsingContextImpl.getTimestamp();
         this.#url = params.frame.url + (params.frame.urlFragment ?? '');
 
         // At the point the page is initialized, all the nested iframes from the
         // previous page are detached and realms are destroyed.
         // Remove children from context.
         this.#deleteAllChildren();
+
+        this.#eventManager.registerEvent(
```

Comment:
After some CDP log dumpster diving. It seems this event is emitted before the `Page.frameStartedLoading`